service/worker/migration/activities.go

```diff
@@ -567,44 +538,67 @@ func (a *activities) verifyHandleNotFoundWorkflow(
 
 	if err != nil {
 		if isNotFoundServiceError(err) {
-			// Workflow could be deleted due to retention.
-			result.Status = VERIFY_SKIPPED
-			result.Reason = reasonWorkflowNotFound
-			return nil
+			// In a rare case, workflow could be deleted due to retention and never being replicated to target.
+			a.forceReplicationMetricsHandler.Counter(metrics.EncounterNotFoundWorkflowCount.GetMetricName()).Record(1)
+			a.logger.Info("createReplicationTasks skip NotFound workflow", tags...)
+			return true, reasonWorkflowNotFound, nil
 		}
 
-		return err
+		return false, "", err
 	}
 
+	// Zombie workflow should be a transient state. However, if there is Zombie workflow on the source cluster,
```

Comment:
We might need to revisit this when we do GN and migration. This is not true any more for GN.